client/app/plan-builder/plan-builder.controller.js

```diff
@@ -36,13 +35,66 @@ angular.module('prosperenceApp')
   };
   updateRelationals($state.current);
 
+  // Checks each query object in the current queries object for completeness. Returns boolean.
+  $scope.checkQueriesComplete = function(queries, plangroup) {
+    if (!!queries) {
+      for (var i=0, n=queries.length; i<n; i++) {
+        queries[i].isComplete = checkQueryComplete(queries[i], plangroup);
+        if (queries[i].isComplete && queries[i+1]) {
+          queries[i+1].isEnabled = true;
+        }
+      }
+    }
+  };
+
```

Comment:
Like the named functions not attached to scope here. I've read this is a good habit to get into when AngularJS. 1) Named functions help when debugging, and 2) only putting what functions / variables necessarily must be on $scope reduces clutter on $scope and could help in performance (less dirty checking?).